holoviews/plotting/bokeh/links.py

```diff
@@ -151,8 +151,10 @@ def __init__(self, root_model, link, source_plot, target_plot):
             start, end = bounds
             if start is not None:
                 axes[f'{axis}_range'].start = start
+                axes[f'{axis}_range'].reset_start = start
             if end is not None:
                 axes[f'{axis}_range'].end = end
+                axes[f'{axis}_range'].reset_end = end
```

Comment:
I think we should add a option to revert to previous behavior, like:

```suggestion
            if start is not None:
                axes[f'{axis}_range'].start = start
            if end is not None:
                axes[f'{axis}_range'].end = end
            if link.reset_to_bounds:
                axes[f'{axis}_range'].reset_start = start
                axes[f'{axis}_range'].reset_end = end
```

and then add the following in holoviews/plotting/links.py - RangeToolLink class

```
    reset_to_bounds = param.Boolean(default=True, doc="""
        If True (default), the 'reset' action returns the plot to its initial bounds.
        If False, it uses Bokeh's default reset behavior.""")`
```